packages/block-library/src/details/edit.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	InnerBlocks,
+	RichText,
+	InspectorControls,
+} from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { ToggleControl } from '@wordpress/components';
+import { useEffect, useRef } from '@wordpress/element';
+import { SPACE } from '@wordpress/keycodes';
+
+export default ( {
+	attributes,
+	className,
+	clientId,
+	isSelected,
+	setAttributes,
+} ) => {
+	const summaryRef = useRef( null );
+
+	useEffect( () => {
+		if ( ! summaryRef.current ) {
+			return;
+		}
+
+		const keyDownListener = ( e ) => {
+			if ( e.keyCode === SPACE ) {
+				e.preventDefault();
+			}
+		};
+
+		const clickListener = ( e ) => e.preventDefault();
+
+		summaryRef.current.addEventListener( 'keyup', keyDownListener );
+		summaryRef.current.addEventListener( 'click', clickListener );
+		return () => {
+			summaryRef.current.removeEventListener( 'keyup', keyDownListener );
+			summaryRef.current.removeEventListener( 'click', clickListener );
+		};
+	}, [ summaryRef.current ] );
+
+	const isInnerBlockSelected = useSelect(
+		( select ) =>
+			select( 'core/block-editor' ).hasSelectedInnerBlock( clientId ),
+		[ clientId ]
+	);
+
+	const showInnerBlocks =
+		attributes.initialOpen || isSelected || isInnerBlockSelected;
+
+	return (
+		<>
+			<InspectorControls>
+				<ToggleControl
+					label={ __( 'Open by default' ) }
+					onChange={ ( initialOpen ) =>
+						setAttributes( { initialOpen } )
+					}
+					checked={ attributes.initialOpen }
+				/>
+			</InspectorControls>
+			<details className={ className } open={ showInnerBlocks }>
+				<summary ref={ summaryRef }>
+					<RichText
```

Comment:
I wonder, could the tag name of rich text be `summary`?

Reply:
I tried that initially but the `summary` was swallowing the space key event and so everywordwouldruntogether. Making `RichText` a child of `summary` fixed that issue in Chrome and Safari, but not in Firefox.

The other approach I took was to use divs and CSS to emulate `details` in the editor view, and that worked, but I would much rather use the native elements if possible.

Reply:
@ellatrix Do you know if there is a way to tell `RichText` to insert a [space] at the current caret position? I'm going to poke around the components in `@wordpress/rich-text` and `@wordpresss/block-editor` to see if that would allow us to continue using `<summary>` in Firefox.

Reply:
I wound up making a change to the `RichText` component that fixed it in Firefox and allowed us to use `summary` as our `tagName`